Tom/Classes/TomView.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  TomView.swift
+//  Tom
+//
+//  Created by Yusuf Demirci on 30.03.21.
+//
+
+final public class TomView: UIView {
+
+    // MARK: Properties
+
+    private let stackView: UIStackView = {
+        let view: UIStackView = .init()
+        view.distribution = .fillEqually
+        return view
+    }()
+
+    private var lineViews: [LineView] = []
+    private var isAnimationEnabled: Bool = false
+
+    private let configuration: Configuration
+
+    public init(configuration: Configuration, frame: CGRect = .init()) {
+        self.configuration = configuration
+
+        super.init(frame: frame)
+
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Publics
+
+public extension TomView {
+
+    func start() {
+        isAnimationEnabled = true
+
+        lineViews.forEach { [weak self] lineView in
+            self?.animate(lineView: lineView)
+        }
+    }
+
+    func stop() {
+        isAnimationEnabled = false
+
+        lineViews.forEach { [weak self] lineView in
+            lineView.heightConstraint?.constant = self?.configuration.minLineHeight ?? 0
+
+            UIView.animate(withDuration: configuration.animationSpeed) {
+                self?.layoutIfNeeded()
+            }
+        }
+    }
+}
+
+// MARK: - Privates
+
+private extension TomView {
+
+    func setup() {
+        stackView.spacing = configuration.lineSpacing
+        stackView.axis = configuration.axis
+
+        self.addSubview(stackView)
+        stackView.pinToView(to: self)
+
+        addLinesToStackView()
+    }
+
+    func animate(lineView: UIView) {
+        guard let maxHeight = lineView.superview?.frame.height else { return }
+
+        let newHeight = CGFloat.random(in: configuration.minLineHeight...maxHeight)
+        lineView.heightConstraint?.constant = newHeight
+
+        UIView.animate(withDuration: configuration.animationSpeed) { [weak self] in
+            self?.layoutIfNeeded()
+        } completion: { [weak self] _ in
+            if self?.isAnimationEnabled ?? false {
```

Comment:
Can we unwrap the optional "self" with "guard" statement here? 

`guard let self = self else { return }`